mitodl/mitxpro
# Check for Hubspot API errors

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #328

#### What's this PR do?
- Adds a celery task to check for any new Hubspot API errors and log them.
- Schedules the task to run every X minutes (determined by `HUBSPOT_ERROR_CHECK_FREQUENCY` setting or 600 seconds by default)
- Adds a `HubspotErrorTimestamp` model to track the last time errors were checked.

#### How should this be manually tested?
- Set `HUBSPOT_ERROR_CHECK_FREQUENCY=60`.  Within a minute, a `HubspotErrorTimestamp` should be created with the current date/time, but no Hubspot API errors should be logged.
- Manually modify the HubspotErrorTimestamp's `checked_on` value to be sometime before May 2019.
- Wait a minute again.  A bunch of Hubspot errors should show up in the log.  Afterward, check that there is still only one `HubspotErrorCheck` object and it's `checked_on` date is close to the current date/time.
- Wait another few minutes, the errors that were previously logged should not be logged again.
